### PR TITLE
[azure] Add max sockets configuration

### PR DESCRIPTION
**Pull request by mattsp1290**
### What does this PR do?

Adds a max sockets configuration and defaults to standard Azure Function App limits. 

### Testing Guidelines

Has been tested on a function app deployed in Azure configured to send logs to Datadog.

### Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)
- [ ] This PR passes the unit tests 
- [ ] This PR passes the installation tests (ask a Datadog member to run the tests)

### CODE REVIEW

File: azure/activity_logs_monitoring/index.js [hunk 1]
```diff
@@ -31,6 +31,8 @@ const DD_PARSE_DEFENDER_LOGS = process.env.DD_PARSE_DEFENDER_LOGS; // Boolean wh
 
 const MAX_RETRIES = 4; // max number of times to retry a single http request
 const RETRY_INTERVAL = 250; // amount of time (milliseconds) to wait before retrying request, doubles after every retry
+const MAX_SOCKETS = Number(process.env.MAX_SOCKETS || 128); // max number of sockets to use for http requests
```
Comment (parsons90) [inline azure/activity_logs_monitoring/index.js]: Are there Azure docs that show the default is 128? I tried searching for some, but couldn't find any.

**Review by parsons90 — COMMENTED** [review]
Question about the default

File: azure/activity_logs_monitoring/index.js (hunk 1 above)
Comment (mattsp1290) [reply azure/activity_logs_monitoring/index.js]: https://4lowtherabbit.github.io/blogs/2019/10/SNAT/#snat-port-allocation-algorithms

Comment (mattsp1290) [reply azure/activity_logs_monitoring/index.js]: Links back to this https://azure.microsoft.com/en-us/blog/azure-load-balancer-to-become-more-efficient/
Seems like 1024 is the absolute max. Will change the default to that to ensure we do not degrade any customers

Comment (parsons90) [reply azure/activity_logs_monitoring/index.js]: I found this: https://learn.microsoft.com/en-us/azure/azure-functions/manage-connections?tabs=javascript#:~:text=The%20global%20http.globalAgent%20class%20used%20by%20http.request()%20has%20all%20of%20these%20values%20set%20to%20their%20respective%20defaults.
and
https://nodejs.org/dist/latest-v6.x/docs/api/http.html#http_agent_maxsockets

Comment (parsons90) [reply azure/activity_logs_monitoring/index.js]: These seem to be 2 different values to configure

Comment (parsons90) [reply azure/activity_logs_monitoring/index.js]: Or I don't understand the issue that's trying to be solved

Comment (mattsp1290) [reply azure/activity_logs_monitoring/index.js]: The log forwarder changes are coming from here https://datadoghq.atlassian.net/browse/CLOUDS-4476
Each invocation of a function app has a maximum amount of pre-allocation ports on the load balancer. If they exceed that amount they will start having to wait for new connections. The max sockets setting forces our request fan to stay within those limits.

**Review by parsons90 — APPROVED** [review]
nit: it would be nice to have a link to the documentation that shows the default value as a comment in the forwarder

**Review by parsons90 — APPROVED** [review]

**Pull request merged**